### PR TITLE
Site: Search with DocSearch by @Algolia

### LINKED ISSUE #6508: Site: add search
```gherkin
As a user
I am able to search Jekyll documentation
```

This is easy to implement with [Docsearch by Algolia](https://github.com/algolia/docsearch)

### PR DESCRIPTION
**Pull request by jekyllbot**
Fix #6508 

This PR add search to Jekyll's website thanks to @Algolia's [DocSearch](https://community.algolia.com/docsearch/)

[Configuration is versioned](https://github.com/algolia/docsearch-configs/blob/master/configs/jekyllrb.json) on Algolia's repository

### CODE REVIEW

File: docs/latest_version.txt [hunk 1]
```diff
@@ -1 +1 @@
-3.6.2
+3.6.0
```
Comment (ashmaroli) [inline docs/latest_version.txt]: This change can be removed..

File: docs/_sass/_docsearch.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.algolia-autocomplete .algolia-docsearch-suggestion--highlight {
+  background-color: #fc0;
+  color: #222;
+}
+
+.algolia-autocomplete .algolia-docsearch-suggestion--text .algolia-docsearch-suggestion--highlight {
+  box-shadow: inset 0 -2px 0 0 #fc0;
+}
```
Comment (ashmaroli) [inline docs/_sass/_docsearch.scss]: if this is not an auto-generated page, can we have these nested?
```sass
.algolia-autocomplete {
  .algolia-docsearch-suggestion--category-header { }
  .algolia-docsearch-suggestion--subcategory-column { }
}
```

**Comment by pup** [comment]
@DirtyF any way to vertically align the search bar with the other menu items?

image

**Comment by pup** [comment]
also, this 🤔 

image

File: docs/_includes/search/input.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<input type="text" id="docsearch-input" placeholder="Search the docs…">
```
Comment (pup) [inline docs/_includes/search/input.html]: is there any usecase for including this somewhere else?

**Comment by DirtyF** [comment]
@pup fixed thanks to @s-pace in https://github.com/algolia/docsearch-configs/commit/9f0d791827ae233d40934e1530cc4cf6a142caaf 🔧 

capture d ecran 2017-11-14 20 12 35

🎉

**Comment by s-pace** [comment]
Feel free to ping us.
Would be glad to help you and enhance this search experience!
Well done, thanks @DirtyF

**Comment by DirtyF** [comment]
@s-pace the last thing I would like to customize is the light blue background color, and I think we're good to go.

**Comment by DirtyF** [comment]
@pup fixed the positioning

search-position

**Comment by DirtyF** [comment]
Styles are OK now

jekyll-search

File: docs/latest_version.txt (hunk 1 above)
Comment (DirtyF) [reply docs/latest_version.txt]: fixed

File: docs/_sass/_docsearch.scss (hunk 2 above)
Comment (DirtyF) [reply docs/_sass/_docsearch.scss]: fixed

File: docs/_includes/search/input.html (hunk 3 above)
Comment (DirtyF) [reply docs/_includes/search/input.html]: I just wanted to isolate the component in its own file

File: docs/_includes/search/script.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+<script type="text/javascript"> docsearch({
+apiKey: '50fe39c839958dfad797000f33e2ec17',
+indexName: 'jekyllrb',
+inputSelector: '#docsearch-input',
+enhancedSearchInput: true,
+debug: true // Set debug to true if you want to inspect the dropdown
```
Comment (DirtyF) [inline docs/_includes/search/script.html]: I'll set this to false before merging.

File: docs/_sass/_docsearch.scss [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }
+}
+
+.algolia-autocomplete {
+  .ds-dropdown-menu {
+    font-size: 1rem;
+    text-shadow: none;
+
+    .ds-suggestion.ds-cursor .algolia-docsearch-suggestion:not(.suggestion-layout-simple) .algolia-docsearch-suggestion--content {
```
Comment (ashmaroli) [inline docs/_sass/_docsearch.scss]: I guess this line can be shortened by further nesting.. no? :smiley:

**Comment by DirtyF** [comment]
Let's get this live and see how it behaves. We mostly need feedback from users.

@jekyllbot: merge +site

**Comment by s-pace** [comment]
Regarding those feedbacks, I have granted @DirtyF to the feedback we have from your search bar.

Feel free to ask me to add other people by sending me the email address.

Congrats

**Pull request merged**